pallets/clients-info/src/lib.rs

```diff
@@ -108,22 +129,73 @@ pub mod pallet {
 			client_name: NameOf<T>,
 			release: ClientRelease<UriOf<T>, T::Hash>,
 		) -> DispatchResult {
-			ensure_root(origin)?;
+			Pallet::<T>::check_origin_rights(origin)?;
 			PendingClientReleases::<T>::insert(client_name, release.clone());
 			Self::deposit_event(Event::<T>::NotifyClientRelease { release });
 			Ok(())
 		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(<T as Config>::WeightInfo::authorize_account())]
+		#[transactional]
+		pub fn authorize_account(origin: OriginFor<T>, account_id: T::AccountId) -> DispatchResult {
+			Pallet::<T>::check_origin_rights(origin)?;
+
+			if !<AuthorizedAccounts<T>>::contains_key(&account_id) {
+				Self::deposit_event(Event::<T>::AccountIdAuthorized(account_id.clone()));
+				<AuthorizedAccounts<T>>::insert(account_id, ());
+			}
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(<T as Config>::WeightInfo::deauthorize_account())]
+		#[transactional]
+		pub fn deauthorize_account(
+			origin: OriginFor<T>,
+			account_id: T::AccountId,
+		) -> DispatchResult {
+			if let Ok(origin_account_id) = ensure_signed(origin.clone()) {
+				Pallet::<T>::check_non_root_rights(origin.clone())?;
```

Comment:
How about we change the return type of `check_non_root_rights()` to return the `AccountId` each time? Then we can get rid of the `if let` in the previous line.

Reply:
Also, with the currnt logic, anyone can call this extrinsic unsigned and we will skip this if statement completely and go to the next if statement where we remove. This is a flaw that we would fix with my suggested change.

Reply:
`check_non_root_rights()` already returns the account id.

Reply:
> check_non_root_rights() already returns the account id.

True, sorry. I actually was referring to `check_origin_rights()`.